src/Http/Controllers/EloquentController.php

```diff
@@ -244,7 +244,45 @@ protected function hydrate(ResourceInterface $resource, Model $model)
      */
     protected function commit(Model $model)
     {
-        return $model->save();
+        $isUpdating = $model->exists;
+
+        $this->beforeCommit($model, $isUpdating);
+
+        $result = $model->save();
+
+        $this->afterCommit($model, $isUpdating);
+
+        return $result;
+    }
+
+    /**
+     * Determines which callback to use before creating or updating a model.
+     *
+     * @param Model $model
+     * @param bool $isUpdating
+     */
+    protected function beforeCommit(Model $model, $isUpdating)
+    {
+        if ($isUpdating) {
+            return $this->updating($model);
+        }
+
+        return $this->creating($model);
```

Comment:
I'd prefer (but feel free to disagree!) that `return` wasn't used in this method, because the return value isn't actually going to be used.